File: account/regional-settings/get-location-list/code-samples/listLocations.cs
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// OPTIONAL QUERY PARAMETERS
ListLocationsParameters listLocationsParameters = new ListLocationsParameters {
    //orderBy = 'City',
    //page = 1,
    //perPage = 100,
    //stateId = "<ENTER VALUE>",
    //withNxx = true
};

RestClient rc = new RestClient(
    Environment.GetEnvironmentVariable("clientId"),
    Environment.GetEnvironmentVariable("clientSecret"),
    false
);
await rc.Authorize(
    Environment.GetEnvironmentVariable("username"),
    Environment.GetEnvironmentVariable("extension"),
    Environment.GetEnvironmentVariable("password")
);
var r = await rc.Restapi().Dictionary().Location().Get(listLocationsParameters);
// PROCESS RESPONSE

File: account/regional-settings/get-location-list/code-samples/listLocations.js
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// OPTIONAL QUERY PARAMETERS
const queryParams = {
    //orderBy: 'City',
    //page: 1,
    //perPage: 100,
    //stateId: '<ENTER VALUE>',
    //withNxx: true
};

const SDK = require('ringcentral');
const rcsdk = new SDK({server: process.env.serverURL, appKey: process.env.clientId, appSecret: process.env.clientSecret});
const platform = rcsdk.platform();
platform.login({ username: process.env.username, extension: process.env.extension, password: process.env.password }).then(() => {
    platform.get(`/restapi/v1.0/dictionary/location`, queryParams).then((r) => {
        // PROCESS RESPONSE
    });
});
